### PR TITLE
Remove useInMemory JSP option as it's broken and not requested

### PR DESCRIPTION
**Pull request by volosied**
for #15240 and also #15035

This might need a POC meeting? 

I looked into fixing the in memory option, but it was more trouble it it's worth. No one has mentioned that this option doesn't work and no one seems to be using it.  I think it's better to just remove it.

### CODE REVIEW

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat,io.openliberty.pages.3.1.internal_fat,io.openliberty.pages.4.0.internal_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_lWAS_jsp.2.3,com.ibm.ws.jsp_fat,com.ibm.ws.jsp_bvt,com.ibm.ws.jsp_fat_lWAS.toleration.ee9,com.ibm.ws.jsp_fat_lWAS.toleration.ee10,com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_jsf2.2,com.ibm.ws.jsf_fat_jsf2.3,com.ibm.ws.jsf_fat_lWAS,com.ibm.ws.jsf_fat_lWAS_jsf2.2,com.ibm.ws.jsf_fat_lWAS_servlet3.1,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat,com.ibm.ws.webcontainer_fat

**Comment by volosied** [comment]
Everything passed except for 
image


Just need to ignore the metatype error: `junit.framework.AssertionFailedError: Type com.ibm.ws.jsp.2.2 is missing attribute useInMemory which existed in the baseline with type booleanType.`

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionProcessor.java [hunk 1]
```diff
@@ -458,8 +439,8 @@ public IServletWrapper findWrapper(HttpServletRequest httpReq, HttpServletRespon
 
         if (jspOptions.isDisableJspRuntimeCompilation() == false && // 223399
             (jspOptions.getTranslationContextClass() == null // 225901 
-            || (jspOptions.getTranslationContextClass() != null && // 415289
-            jspOptions.getTranslationContextClass().equals(Constants.IN_MEMORY_TRANSLATION_CONTEXT_CLASS)))) {
+            || jspOptions.getTranslationContextClass() != null // 415289
+            )) {
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/webcontainerext/AbstractJSPExtensionProcessor.java]: Remove the `|| jspOptions.getTranslationContextClass() != null` check.

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java [hunk 2]
```diff
@@ -246,23 +243,14 @@ public static JspTranslationResult translateJsp(
 
         String jspVisitorCollectionId = null;
         String tagFileVisitorCollectionId = null;
-        if (!options.isUseInMemory()) {
-	        jspVisitorCollectionId = JSP_TRANSLATION_ID;
-	        tagFileVisitorCollectionId = TAGFILE_TRANSLATION_ID;
+	    
+        jspVisitorCollectionId = JSP_TRANSLATION_ID;
+	    tagFileVisitorCollectionId = TAGFILE_TRANSLATION_ID;
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java]: formatting

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/taglib/TagLibraryCache.java [hunk 3]
```diff
@@ -222,8 +219,8 @@ private TagLibraryCache(JspTranslationContext ctxt,
 
         if (jspOptions.isUseImplicitTagLibs() &&
              (jspOptions.getTranslationContextClass() == null 
-              	|| (jspOptions.getTranslationContextClass() != null && // 415289
-              	    jspOptions.getTranslationContextClass().equals(Constants.IN_MEMORY_TRANSLATION_CONTEXT_CLASS)))) { 
+              	|| jspOptions.getTranslationContextClass() != null // 415289
```
Comment (volosied) [inline dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/taglib/TagLibraryCache.java]: Fix up this check. Remove `|| jspOptions.getTranslationContextClass() != null` check

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat,io.openliberty.pages.3.1.internal_fat,io.openliberty.pages.4.0.internal_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_lWAS_jsp.2.3,com.ibm.ws.jsp_fat,com.ibm.ws.jsp_bvt,com.ibm.ws.jsp_fat_lWAS.toleration.ee9,com.ibm.ws.jsp_fat_lWAS.toleration.ee10,com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_jsf2.2,com.ibm.ws.jsf_fat_jsf2.3,com.ibm.ws.jsf_fat_lWAS,com.ibm.ws.jsf_fat_lWAS_jsf2.2,com.ibm.ws.jsf_fat_lWAS_servlet3.1,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat,com.ibm.ws.webcontainer_fat

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
New Build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_Rh9U4GyGEe6Qs_3uoOzbQw

**Comment by volosied** [comment]
New build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_rWLaEHHaEe6LyICjST0JEw

**Review by epj — DISMISSED** [review]
nice work

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java (hunk 2 above)
Comment (pnicolucci) [reply dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java]: the tagFileVisitorCollectionId still looks to be indented too far?

Comment (pnicolucci) [reply dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java]: Also why are we setting these to null and then immediately giving them a value?

**Review by pnicolucci — COMMENTED** [review]
I've left a couple of minor review comments. Thanks Vlad!

File: dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java (hunk 2 above)
Comment (volosied) [reply dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/utils/JspTranslatorUtil.java]: Ah good catch. Let me fix this

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jsp.2.3_fat,io.openliberty.pages.3.1.internal_fat,io.openliberty.pages.4.0.internal_fat,com.ibm.ws.jsp_fat_lWAS,com.ibm.ws.jsp_fat_lWAS_jsp.2.3,com.ibm.ws.jsp_fat,com.ibm.ws.jsp_bvt,com.ibm.ws.jsp_fat_lWAS.toleration.ee9,com.ibm.ws.jsp_fat_lWAS.toleration.ee10,com.ibm.ws.jsf_fat,com.ibm.ws.jsf_fat_jsf2.2,com.ibm.ws.jsf_fat_jsf2.3,com.ibm.ws.jsf_fat_lWAS,com.ibm.ws.jsf_fat_lWAS_jsf2.2,com.ibm.ws.jsf_fat_lWAS_servlet3.1,com.ibm.ws.jsf.2.2_fat,com.ibm.ws.jsf.2.3_fat,com.ibm.ws.webcontainer_fat

**Review by pnicolucci — APPROVED** [review]

**Comment by volosied** [comment]
#libby

**Pull request merged**